src/unistd.rs

```diff
@@ -20,12 +20,22 @@ pub use self::pivot_root::*;
 #[cfg(any(target_os = "android", target_os = "freebsd",
           target_os = "linux", target_os = "openbsd"))]
 pub use self::setres::*;
+#[cfg(not(any(target_os = "android", 
```

Comment:
Why did you have to exclude so many architectures?  Shouldn't this be architecture independent?